src/domain/sales-channels/tables/product.tsx

```diff
@@ -0,0 +1,486 @@
+  }
+
+  return (
+    <div
+      className={clsx(
+        "absolute w-full bottom-1 flex justify-center transition-all duration-200 pointer-events-none",
+        classes
+      )}
+    >
+      <div className="h-[48px] min-w-[224px] rounded-lg border shadow-toaster flex items-center justify-around gap-3 px-4 py-3 pointer-events-auto">
+        <span className="text-small text-grey-50">{total} selected</span>
+        <div className="w-[1px] h-[20px] bg-grey-20" />
+        <Button variant="danger" size="small" onClick={onRemove}>
+          Remove
+        </Button>
+        <button onClick={close} className="text-grey-50 cursor-pointer">
+          <CrossIcon size={20} />
+        </button>
+      </div>
+    </div>
+  )
+}
+
+/* **************************************** */
+/* ************** CONTAINERS ************** */
+/* **************************************** */
+
+type SalesChannelProductsTableProps = {
+  salesChannelId: string
+  showAddModal: () => void
+}
+
+/**
+ * Sales channel products table container.
+ */
+function SalesChannelProductsTable(props: SalesChannelProductsTableProps) {
+  const { salesChannelId, showAddModal } = props
+  const [selectedRowIds, setSelectedRowIds] = useState([])
+
+  const notification = useNotification()
+
+  const params = useQueryFilters(defaultQueryProps)
+  const filters = useProductFilters()
+
+  const {
+    mutate: deleteProductsFromSalesChannel,
+  } = useAdminDeleteProductsFromSalesChannel(salesChannelId)
+
+  const { products, count, isLoading } = useAdminProducts({
+    ...params.queryObject,
+    ...filters.queryObject,
+  })
+
+  const removeProductFromSalesChannel = (id: string) => {
+    deleteProductsFromSalesChannel({ product_ids: [{ id }] })
+
+    notification("Success", "Product successfully removed", "success")
+  }
+
+  const removeSelectedProducts = async () => {
+    await deleteProductsFromSalesChannel({
+      product_ids: selectedRowIds.map((id) => ({ id })),
+    })
+
+    notification(
+      "Success",
+      "Products successfully removed from the sales channel",
+      "success"
+    )
+    setSelectedRowIds([])
+  }
+
+  // TODO: use the products endpoint to do this once `expand` with `sales_channels` is supported
+  // until then proper pagination or revalidation of data on products select wont work
+  const filteredProducts = useMemo(
+    () =>
+      products?.filter(
+        (product) =>
+          !!product.sales_channels!.find(
+            (channel) => channel.id === salesChannelId
+          )
+      ),
+    [products, salesChannelId]
+  )
+
+  const isFilterOn = Object.keys(filters.queryObject).length
+  const hasSearchTerm = params.queryObject.q
+
+  if (
+    !filteredProducts?.length &&
+    !isLoading &&
+    !isFilterOn &&
+    !hasSearchTerm
+  ) {
+    return <Placeholder showAddModal={showAddModal} />
+  }
+
+  const toBeRemoveCount = selectedRowIds.length
+
+  return (
+    <div className="relative h-[880px]">
+      <ProductTable
```

Comment:
Table doesn't refresh when removing products (the expand pr has been merged)

Reply:
added proper cache keys with [this PR](https://github.com/medusajs/medusa/pull/1931/files), now the list is refreshed properly